src/dex/solidly/forks-override/pharaohV1.ts

```diff
@@ -0,0 +1,253 @@
+import { Network, SUBGRAPH_TIMEOUT } from '../../../constants';
+import { getDexKeysWithNetwork } from '../../../utils';
+import { SolidlyConfig } from '../config';
+import _ from 'lodash';
+import { Address, PoolLiquidity } from '../../../types';
+import BigNumber from 'bignumber.js';
+import { Solidly } from '../solidly';
+import { IDexHelper } from '../../../dex-helper';
+import { SolidlyPair } from '../types';
+import { Interface } from '@ethersproject/abi';
+
+const PharaohV1FactoryABI = [
+  {
+    inputs: [{ internalType: 'address', name: '', type: 'address' }],
+    name: 'pairFee',
+    outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+const pharaohV1FactoryIface = new Interface(PharaohV1FactoryABI);
+
+export type PharaohV1SubgraphPool = {
+  id: string;
+  isStable: boolean;
+  token0: string;
+  reserve0: string;
+  reserve1: string;
+  token1: string;
+};
+
+export type PharaohV1SubgraphToken = {
+  id: string;
+  decimals: string;
+};
+
+export class PharaohV1 extends Solidly {
```

Comment:
@JackieJoo I think we don't need this separate implementation since it's a copy paste from [Ramses class](https://github.com/paraswap/paraswap-dex-lib/blob/feat/BACK-1532/src/dex/solidly/forks-override/ramses.ts#L38). Let's extend PharaohV1 from Ramses.

Reply:
done, thx